components/alert/alert-toast.js

```diff
@@ -97,8 +104,20 @@ class AlertToast extends LitElement {
 
 	render() {
 		return html`
-			<div class="d2l-alert-toast-container" data-state="${this._state}" @transitionend=${this._onTransitionEnd}>
-				<d2l-alert type="${ifDefined(this.type)}" @d2l-alert-closed=${this._onCloseClicked} button-text="${ifDefined(this.buttonText)}" ?has-close-button="${!this.hideCloseButton}" subtext="${ifDefined(this.subtext)}">
+			<div
+				class="d2l-alert-toast-container"
+				data-state="${this._state}"
+				@transitionend=${this._onTransitionEnd}>
+				<d2l-alert
+					@blur=${this._closeTimerStart}
+					button-text="${ifDefined(this.buttonText)}"
+					@d2l-alert-closed=${this._onCloseClicked}
+					@focus=${this._closeTimerStop}
```

Comment:
Another slight hick-up I noticed: if the toast has focus, we never want it to close. Currently you can open it, focus on it, then move your mouse away and it'll close.

A good solution to this might be to track whether focus is inside or not (using the focus and blur events) and then in `_closeTimerStop` just bail if focus is still inside.

Reply:
How do we approach the situation where you move focus and still have the mouse overtop? I think I'd have to track `hasMouseover` and `hasFocus` and make sure it's got neither before it starts closing if these are equivalent prompts to keep it open.

Reply:
Yep you're right, you'd need to track both. Fun times!